_posts/2020-08-19-#19509.md

```diff
@@ -0,0 +1,53 @@
+
+* The purpose of this feature is twofold: it should assist people in debugging their node, and it should help people who are unfamiliar with the Bitcoin network to get a feel of how everything works.
+
+* This feature is locked behind a debug-only `-logmessages` option.  When enabled, it writes every message the node recieves to a file, in a folder structure described in the PR.  It uses our `CAutoFile` [here](https://github.com/bitcoin/bitcoin/blob/ac953a3773aa5e940976a79276894f5e892d1c55/src/net.cpp#L2892).
+
+* The log parser lives in a new folder, at `contrib/message-logging/message-logging-parser.py`.  As discussed [here](https://github.com/bitcoin/bitcoin/pull/19509#issuecomment-664374225) it takes a list of files to parse, and optionally an output argument.
+
+* `LogMessage` takes the newly introduced `Span`, as discussed in a [previous PR review club](https://bitcoincore.reviews/18468).
+
+* [This commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/ac953a3773aa5e940976a79276894f5e892d1c55) accounts for the increase in file descriptor usage.  This is due to be improved in [Issue 18911](https://github.com/bitcoin/bitcoin/issues/18911).
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+   (You're always encouraged to put your PR review on GitHub, even after the PR has been merged.)
+
+2. How did you test this PR?  Was setting up and running everything straightforward enough?  Reminder that the test runner has a `--nocleanup` option.
+
+3. Did you verify that file descriptor accounting was being done correctly in [this commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/ac953a3773aa5e940976a79276894f5e892d1c55)?  What is the soft file desciptor limit on your system?  What about the hard limit?  (TEMP NOTE: I'm happy to lead them into `ulimit` here in the notes, not sure where to draw the line.  Don't want people just blindly running commands)
+
+4. [The first commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/b3a0e528d1128ee456258f4b265b52b7e8fbc6a6) is a cleanup.  Is this justified or is this just noise?  When should cleanups like this be done?
+
+5. [This commit](https://github.com/bitcoin/bitcoin/pull/19509/commits/4fc2e3906377991a05f78a938c0aa2f04056b81c) adds a test.  What is being tested?  What isn't being tested?  Should this be expanded?  Did you read the out-of-tree build discussion [here](https://github.com/bitcoin/bitcoin/pull/19509#discussion_r465770624)?
+
+6. Both [jnewbery](https://github.com/bitcoin/bitcoin/pull/19509#pullrequestreview-447894982) and [practicalswift](https://github.com/bitcoin/bitcoin/pull/19509#pullrequestreview-448269477) proposed extensions to this message logging. What are they? Do you agree that these are a good idea? Are there any extensions that you would like to see?
```

Comment:
suggest jnewbery [comment](url) (same for practicalswift) instead to make clear that the links are to comments and not to the github profiles.

Reply:
done